### PR TITLE
JAMES-2904 Configuration and documentation enhancements

### PR DESCRIPTION
**Pull request by chibenwa**
Origins from remarks made on https://github.com/linagora/james-project/pull/2730

### CODE REVIEW

File: dockerfiles/run/guice/cassandra-ldap/destination/conf/lmtpserver.xml [hunk 1]
```diff
@@ -18,6 +18,8 @@
   under the License.                                           
  -->
 
+<!-- Read https://james.apache.org/server/config-lmtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-ldap/destination/conf/lmtpserver.xml]: dead link?

File: dockerfiles/run/guice/cassandra-ldap/destination/conf/smtpserver.xml [hunk 2]
```diff
@@ -19,6 +19,8 @@
   under the License.
  -->
 
+<!-- Read https://james.apache.org/server/config-smtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-ldap/destination/conf/smtpserver.xml]: dead link?

File: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/lmtpserver.xml [hunk 3]
```diff
@@ -18,6 +18,8 @@
   under the License.                                           
  -->
 
+<!-- Read https://james.apache.org/server/config-lmtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/lmtpserver.xml]: dead link

File: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/smtpserver.xml [hunk 4]
```diff
@@ -19,6 +19,8 @@
   under the License.
  -->
 
+<!-- Read https://james.apache.org/server/config-smtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/smtpserver.xml]: dead link

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/lmtpserver.xml [hunk 5]
```diff
@@ -18,6 +18,8 @@
   under the License.                                           
  -->
 
+<!-- Read https://james.apache.org/server/config-lmtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/lmtpserver.xml]: dead link

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/smtpserver.xml [hunk 6]
```diff
@@ -19,6 +19,8 @@
   under the License.
  -->
 
+<!-- Read https://james.apache.org/server/config-smtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/smtpserver.xml]: dead link

File: dockerfiles/run/guice/cassandra/destination/conf/lmtpserver.xml [hunk 7]
```diff
@@ -18,6 +18,8 @@
   under the License.                                           
  -->
 
+<!-- Read https://james.apache.org/server/config-lmtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra/destination/conf/lmtpserver.xml]: dead link

File: dockerfiles/run/guice/cassandra/destination/conf/smtpserver.xml [hunk 8]
```diff
@@ -19,6 +19,8 @@
   under the License.
  -->
 
+<!-- Read https://james.apache.org/server/config-smtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra/destination/conf/smtpserver.xml]: dead link

File: dockerfiles/run/guice/jpa-smtp/destination/conf/smtpserver.xml [hunk 9]
```diff
@@ -19,6 +19,8 @@
   under the License.
  -->
 
+<!-- Read https://james.apache.org/server/config-smtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/jpa-smtp/destination/conf/smtpserver.xml]: dead link

File: dockerfiles/run/guice/jpa/destination/conf/lmtpserver.xml [hunk 10]
```diff
@@ -18,6 +18,8 @@
   under the License.                                           
  -->
 
+<!-- Read https://james.apache.org/server/config-lmtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/jpa/destination/conf/lmtpserver.xml]: dead link

File: dockerfiles/run/guice/jpa/destination/conf/smtpserver.xml [hunk 11]
```diff
@@ -19,6 +19,8 @@
   under the License.
  -->
 
+<!-- Read https://james.apache.org/server/config-smtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/jpa/destination/conf/smtpserver.xml]: dead link

File: dockerfiles/run/guice/memory/destination/conf/smtpserver.xml [hunk 12]
```diff
@@ -19,6 +19,8 @@
   under the License.
  -->
 
+<!-- Read https://james.apache.org/server/config-smtp.html for further details -->
```
Comment (rouazana) [inline dockerfiles/run/guice/memory/destination/conf/smtpserver.xml]: dead link

File: dockerfiles/run/guice/cassandra-ldap/destination/conf/smtpserver.xml (hunk 2 above)
Comment (Arsnael) [reply dockerfiles/run/guice/cassandra-ldap/destination/conf/smtpserver.xml]: Seems to be instead https://james.apache.org/server/config-smtp-lmtp.html ?

File: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/lmtpserver.xml (hunk 3 above)
Comment (Arsnael) [reply dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/lmtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/smtpserver.xml (hunk 4 above)
Comment (Arsnael) [reply dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/smtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/lmtpserver.xml (hunk 5 above)
Comment (Arsnael) [reply dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/lmtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/smtpserver.xml (hunk 6 above)
Comment (Arsnael) [reply dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/smtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/cassandra/destination/conf/lmtpserver.xml (hunk 7 above)
Comment (Arsnael) [reply dockerfiles/run/guice/cassandra/destination/conf/lmtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/cassandra/destination/conf/smtpserver.xml (hunk 8 above)
Comment (Arsnael) [reply dockerfiles/run/guice/cassandra/destination/conf/smtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/jpa-smtp/destination/conf/smtpserver.xml (hunk 9 above)
Comment (Arsnael) [reply dockerfiles/run/guice/jpa-smtp/destination/conf/smtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/jpa/destination/conf/lmtpserver.xml (hunk 10 above)
Comment (Arsnael) [reply dockerfiles/run/guice/jpa/destination/conf/lmtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/memory/destination/conf/smtpserver.xml (hunk 12 above)
Comment (Arsnael) [reply dockerfiles/run/guice/memory/destination/conf/smtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: dockerfiles/run/guice/jpa/destination/conf/smtpserver.xml (hunk 11 above)
Comment (Arsnael) [reply dockerfiles/run/guice/jpa/destination/conf/smtpserver.xml]: https://james.apache.org/server/config-smtp-lmtp.html

File: src/site/xdoc/server/config-cassandra.xml [hunk 13]
```diff
@@ -206,6 +206,21 @@
...
 
+    <subsection name="From V5 to V6">
+
+      <p>Goal is to no longer rely on an UDT partition key for mailboxPath tables. Entries will be migrated to mailboxPathV2 table relying on a composite primary key</p>
+
+    </subsection>
+
+    <subsection name="From V6 to V7">
+
+      <p>Goal is to populate mapping_sources projection table. This tables allows finding the source of a given redirection, which is
```
Comment (Arsnael) [inline src/site/xdoc/server/config-cassandra.xml]: s/tables/table

**Comment by Arsnael** [comment]
> we have an other issue: all your link are pointing to master documentation, while they should be pointing to specific documentation version for the concerned release. But this is not in this PR that we will fix that...

I don't think we have at all a versioned website documentation at the moment right? I agree that would be necessary at some point, but pretty sure we miss more or less previous versions documentations as we just keep patching up stuff?

File: dockerfiles/run/guice/cassandra-ldap/destination/conf/lmtpserver.xml (hunk 1 above)
Comment (rouazana) [reply dockerfiles/run/guice/cassandra-ldap/destination/conf/lmtpserver.xml]: more https://james.apache.org/server/config-smtp-lmtp.html#LMTP_Configuration

Comment (Arsnael) [reply dockerfiles/run/guice/cassandra-ldap/destination/conf/lmtpserver.xml]: Yep better with the tag indeed (same goes for smtp)

File: dockerfiles/run/guice/memory/destination/conf/elasticsearch.properties [hunk 14]
```diff
@@ -17,6 +17,8 @@
 
 #  This template file can be used as example for James Server configuration
 #  DO NOT USE IT AS SUCH AND ADAPT IT TO YOUR NEEDS
+# Read https://james.apache.org/server/config-elasticsearch.html for further details (metrics only)
```
Comment (remk) [inline dockerfiles/run/guice/memory/destination/conf/elasticsearch.properties]: It's not really related to this PR but it seems strange to change value in an `elasticsearch.properties` file for a memory product.

**Comment by Arsnael** [comment]
CI restart...
test this please

**Comment by Arsnael** [comment]
Was :green_apple:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**